internal/identity/identitymanager.go

```diff
@@ -146,6 +147,34 @@ func (im *identityManager) NormalizeSigningKey(ctx context.Context, inputKey str
 	return signer.Value, nil
 }
 
+func (im *identityManager) ResolveInputSigningKey(ctx context.Context, inputKey *core.VerifierRef) (*core.VerifierRef, error) {
+	log.L(ctx).Debugf("Resolving input signing key: type='%s' value='%s'", inputKey.Type, inputKey.Value)
+
+	if im.blockchain == nil {
+		return nil, i18n.NewError(ctx, coremsgs.MsgBlockchainNotConfigured)
+	}
+
+	verifierType := inputKey.Type
+
+	if verifierType.String() == "" {
+		verifierType = im.blockchain.VerifierType()
+	}
+
+	if verifierType != im.blockchain.VerifierType() {
```

Comment:
If verifier type is invalid, an error is thrown. I think this is better than using the default blockchain plugin, as it will better inform the user